src/lib/Liblog/chk_file_sec.c

```diff
@@ -116,12 +117,12 @@ teststat(struct stat *sp, int isdir, int sticky, int disallow,
  */
 
 static int
-teststat(struct stat *sp, int isdir, int sticky, int disallow)
+teststat(struct stat *sp, int isdir, int sticky, int disallow, int uid)
 {
 	int	rc = 0;
 
-	if ((~disallow & S_IWUSR) && (sp->st_uid > 10)) {
-		/* Owner write is allowed, and UID is greater than 10. */
+	if ((~disallow & S_IWUSR) && (sp->st_uid > 10 && !(uid > 0 && uid == sp->st_uid))) {
```

Comment:
Yes, because when we check /var/spool/pbs/sched_priv, /var/spool/pbs's owner is root. If i didn't leave this, sched_priv/logs would have to be in directories owned by PBS_DAEMON_SERVICE_USER

Reply:
But isn't it wrong that if st_uid is <=10 then it will not give EPERM error even when daemon service user uid isn't the matching with st_uid?

Reply:
I'm not sure why we trust uids <=10, but we do. I only add the case were we dont fail if `uid == sp->st_uid`.
I really don't see another way around this.

Reply:
Take a gander at your /etc/passwd file.  The UIDs under 10 are system accounts.  It probably was more so 20 years back.